libs/utils/src/sync/gate.rs

```diff
@@ -85,6 +85,13 @@ impl Gate {
         warn_if_stuck(self.do_close(), &self.name, Duration::from_millis(1000)).await
     }
 
+    /// Check if [`Self::close()`] has finished waiting for all [`Self::enter()`] users to finish.  This
+    /// is usually analoguous for "Did shutdown finish?" for types that include a Gate, whereas checking
+    /// the CancellationToken on such types is analogous to "Did shutdown start?"
+    pub fn close_complete(&self) -> bool {
```

Comment:
Note to reviewers: 

Q: Why isn't this just called `is_closed`?
A: Because I want to emphasize the distinction between entering & exiting close.

Reply:
Maybe I'd prefer `is_` prefix here regardless; I don't really understand the emphasis as this can only be closed once, but I don't mind it either.